test/gtest/unit_ActivationDescriptor.cpp

```diff
@@ -0,0 +1,532 @@
+                               const miopen::TensorDescriptor& dy_desc,
+                               ConstData_t dy,
+                               const miopen::TensorDescriptor& x_desc,
+                               ConstData_t x,
+                               const miopen::TensorDescriptor& dx_desc,
+                               Data_t dx)
+    {
+        // Dummy values
+        float alpha = 1.0f;
+        float beta  = 0.0f;
+
+        miopenStatus_t status;
+        status =
+            desc.Backward(handle, &alpha, y_desc, y, dy_desc, dy, x_desc, x, &beta, dx_desc, dx);
+
+        if(status != miopenStatusSuccess)
+            throw std::runtime_error("Backward failed");
+    }
+
+    static void RunGpu(const miopen::ActivationDescriptor& desc,
+                       const tensor<T>& x,
+                       tensor<T>& y,
+                       tensor<T>& dx,
+                       const tensor<T>& dy,
+                       std::promise<void>& forward_ready)
+    {
+        auto handle = miopen::Handle{};
+
+        auto x_dev = handle.Write(x.data);
+        auto y_dev = handle.Write(y.data);
+
+        RunGpuForward(handle, desc, x.desc, x_dev.get(), y.desc, y_dev.get());
+
+        y.data = handle.Read<T>(y_dev, y.data.size());
+        handle.Finish();
+
+        forward_ready.set_value();
+
+        auto dx_dev = handle.Write(dx.data);
+        auto dy_dev = handle.Write(dy.data);
+
+        RunGpuBackward(handle,
+                       desc,
+                       y.desc,
+                       y_dev.get(),
+                       dy.desc,
+                       dy_dev.get(),
+                       x.desc,
+                       x_dev.get(),
+                       dx.desc,
+                       dx_dev.get());
+
+        dx.data = handle.Read<T>(dx_dev, dx.data.size());
+        handle.Finish();
+    }
+
+    static void
+    RunCpuForward(const miopen::ActivationDescriptor& desc, const tensor<T>& x, tensor<T>& y)
+    {
+        miopen::tests::CpuActivationForward(
+            desc.GetMode(), desc.GetAlpha(), desc.GetBeta(), desc.GetGamma(), x, y);
+    }
+
+    static void RunCpuBackward(const miopen::ActivationDescriptor& desc,
+                               const tensor<T>& y,
+                               const tensor<T>& dy,
+                               const tensor<T>& x,
+                               tensor<T>& dx)
+    {
+        miopen::tests::CpuActivationBackward(
+            desc.GetMode(), desc.GetAlpha(), desc.GetBeta(), desc.GetGamma(), y, dy, x, dx);
+    }
+
+    static void RunCpu(const miopen::ActivationDescriptor& desc,
+                       const tensor<T>& x,
+                       tensor<T>& y,
+                       tensor<T>& dx,
+                       const tensor<T>& dy,
+                       std::promise<void>& forward_ready)
+    {
+        RunCpuForward(desc, x, y);
+        forward_ready.set_value();
+        RunCpuBackward(desc, y, dy, x, dx);
+    }
```

Comment:
```suggestion
    static void RunCpu(const miopen::ActivationDescriptor& desc,
                       const tensor<T>& x,
                       tensor<T>& y,
                       tensor<T>& dx,
                       const tensor<T>& dy,
                       std::promise<void>& forward_ready)
    {
        miopen::tests::CpuActivationForward(
            desc.GetMode(), desc.GetAlpha(), desc.GetBeta(), desc.GetGamma(), x, y);
        forward_ready.set_value();
        miopen::tests::CpuActivationBackward(
            desc.GetMode(), desc.GetAlpha(), desc.GetBeta(), desc.GetGamma(), y, dy, x, dx);
    }
```
All those functions are already `Forward` and `Backward`. Why do we need extra wrappers for them?